client/rpcserver/rpcserver_test.go

```diff
@@ -498,8 +527,9 @@ func TestClientMap(t *testing.T) {
 	s, _, shutdown := newTServer(t, true, "", "")
 	conn := new(TConn)
 
+	s.mtx.(*tRWMutex).locked = make(chan struct{})
```

Comment:
We have the `TConn` stub at our disposal.  How about we modify it to give a signal when the `WSLink` is running with it (after `s.clients` is updated with the `wsClient`) like:

```go

type TConn struct {
	msg       []byte
	reads     [][]byte       // data for ReadMessage
	respReady chan []byte    // signal from WriteMessage
	close     chan struct{}  // Close tells ReadMessage to return with error
}

var readTimeout = 10 * time.Second // ReadMessage must not return constantly with nothing

func (c *TConn) ReadMessage() (int, []byte, error) {
	if len(c.reads) > 0 {
		fmt.Println("Reading dummy message from TConn.")
		var read []byte
		// pop front
		read, c.reads = c.reads[0], c.reads[1:]
		return len(read), read, nil
	}

	select {
	case <-c.close: // receive from nil channel blocks, closed receives immediately
		fmt.Println("Close called, ReadMessage returning with error.")
		return 0, nil, fmt.Errorf("closed")
	case <-time.After(readTimeout):
		return 0, nil, fmt.Errorf("read timeout")
	}
}

func (c *TConn) addRead(read []byte) {
	// push back
	c.reads = append(c.reads, read)
}

func (c *TConn) WriteMessage(_ int, msg []byte) error {
	fmt.Println("Writing dummy response to TConn.")
	c.msg = msg
	select {
	case c.respReady <- msg:
	default:
	}
	return nil
}
```

Then here in `TestClientMap`:

```go
func TestClientMap(t *testing.T) {
	s, _, shutdown := newTServer(t, true, "", "")
	resp := make(chan []byte, 1)
	conn := &TConn{respReady: resp}
	// msg.ID == 0 gets an error response, which can be discarded.
	read, _ := json.Marshal(msgjson.Message{ID: 0})
	fmt.Println(string(read))
	conn.addRead(read)

	go s.websocketHandler(conn, "someip")

	// When a response to our dummy message is received, the client should be in
	// RPCServer's client map.
	msg := <-resp
	fmt.Println("Response to dummy message received from server:", string(msg))
...
```

Reply:
Thank you. That's a good answer for this case.